components/x-teaser-timeline/src/lib/transform.js

```diff
@@ -120,52 +187,116 @@ const getItemGroups = ({
-		const index = groups[group].items.findIndex((item) => item.articleIndex === position - 1)
-
+	if (position === 0) {
 		return {
-			group: group,
-			index: index + 1
+			group: 0,
+			index: 0
 		}
 	}
 
-	return {
-		group: 0,
-		index: 0
-	}
+	return groups
+		.map((group, groupIndex) => {
+			const articleIndexInGroup = group.items.findIndex((item) => item.articleIndex === position - 1)
+			if (articleIndexInGroup >= 0) {
+				return {
+					group: groupIndex,
+					index: articleIndexInGroup + 1
+				}
+			}
+		})
+		.filter((a) => a)
+		.pop()
 }
 
-const interleaveAllSlotsWithCustomSlots = (
+/**
+ * Creates a deep copy of an array of GroupOfItems data structure.
+ * @param {Array<GroupOfItems>} groupedItems
+ * @return {Array<GroupOfItems>}
+ */
+const deepCopyGroupedItems = (groupedItems) =>
+	Array.from(groupedItems, (v, k) => ({
+		...groupedItems[k],
+		items: [...groupedItems[k].items]
+	}))
+
+/**
+ * Inserts each custom slot into the appropriate group of items,
+ * so that the custom slot will appear in the expected
+ * position - specified via the positions array arg - in the teaser timeline.
+ * @param {CustomSlotContent} customSlotContentArray
+ * @param {CustomSlotPosition} customSlotPositionArray
+ * @param {Array<GroupOfItems>} itemGroups
+ * @param {Array<Items>} items
+ * @returns {Array<GroupOfItems>}
+ */
+export const interleaveAllSlotsWithCustomSlots = (
 	customSlotContentArray,
 	customSlotPositionArray,
 	itemGroups,
 	items
 ) => {
+	const interleavedItemGroups = deepCopyGroupedItems(itemGroups)
+
 	for (const [index, slotContent] of customSlotContentArray.entries()) {
 		const insertPosition = Math.min(customSlotPositionArray[index], items.length + index)
-		const insert = getGroupAndIndex(itemGroups, insertPosition)
-		const copyOfItems = [...itemGroups[insert.group].items]
+		const insert = getGroupAndIndex(interleavedItemGroups, insertPosition)
 
-		copyOfItems.splice(insert.index, 0, slotContent)
-		itemGroups[insert.group].items = copyOfItems
+		if (insert) {
+			interleavedItemGroups[insert.group].items.splice(insert.index, 0, slotContent)
+		}
 	}
-	return itemGroups
+
+	return interleavedItemGroups
 }
 
+/**
+ * @typedef {Object} buildModelProps
+ * @property {Array<Item>} items
+ * @property {CustomSlotContent} customSlotContent
+ * @property {CustomSlotPosition} customSlotPosition
+ * @property {number} timezoneOffset
+ * @property {string} localTodayDate e.g., '2020-01-14'
+ * @property {string} latestItemsTime e.g., '2020-01-13T10:00:00+00:00'
+ * @property {number} latestItemsAgeHours e.g., 36
+ *
+ * Builds the XTeaserTimeline data model.
+ * The list of news items passed as argument is divided into groups of items,
+ * where each group identifies a specific date when the subset of news was published.
+ * Custom slots - e.g., ads - are also inserted into their expected position.
+ * @param {buildModelProps}
```

Comment:
```suggestion
 * @typedef {Object} BuildModelProps
 * @property {Array<Item>} items
 * @property {CustomSlotContent} customSlotContent
 * @property {CustomSlotPosition} customSlotPosition
 * @property {number} timezoneOffset
 * @property {string} localTodayDate e.g., '2020-01-14'
 * @property {string} latestItemsTime e.g., '2020-01-13T10:00:00+00:00'
 * @property {number} latestItemsAgeHours e.g., 36
 *
 * Builds the XTeaserTimeline data model.
 * The list of news items passed as argument is divided into groups of items,
 * where each group identifies a specific date when the subset of news was published.
 * Custom slots - e.g., ads - are also inserted into their expected position.
 * @param {BuildModelProps} props
```
Small tweak: `@param` definitions need an accompanying reference (here `props`)